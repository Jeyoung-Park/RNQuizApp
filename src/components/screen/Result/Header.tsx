import * as React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';

interface HeaderProps {
	goBack: () => void;
}

const Header = ({ goBack }: HeaderProps) => (
	// eslint-disable-next-line @typescript-eslint/no-use-before-define
	<View style={styles.container}>
		<TouchableOpacity style={styles.backButton} onPress={goBack}>
			<Text>Back</Text>
		</TouchableOpacity>
		<Text>Result</Text>
	</View>
);

export default Header;

const styles = StyleSheet.create({
	container: {
		width: '100%',
		flexDirection: 'row',
		justifyContent: 'center',

		// backgroundColor: 'red',
		paddingVertical: 10,

		borderBottomColor: 'gray',
		borderBottomWidth: 1,
	},
	backButton: {
		position: 'absolute',
		left: 0,
		paddingTop: 10,
		marginLeft: 16,
	},
});
