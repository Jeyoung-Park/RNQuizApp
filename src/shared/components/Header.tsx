import * as React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/AntDesign';

interface HeaderProps {
	goBack: () => void;
	title: string;
	// eslint-disable-next-line react/require-default-props
	goBackIcon: string;
}

const Header = ({ goBack, title, goBackIcon }: HeaderProps) => (
	// eslint-disable-next-line @typescript-eslint/no-use-before-define
	<View style={styles.container}>
		<TouchableOpacity style={styles.backButton} onPress={goBack}>
			{/* <Text>{goBackText}</Text> */}
			<Icon name={goBackIcon} size={24} color="black" />
		</TouchableOpacity>
		<Text style={styles.title}>{title}</Text>
		<View style={styles.backButton} />
	</View>
);

export default Header;

const styles = StyleSheet.create({
	container: {
		width: '100%',
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		// backgroundColor: 'red',
		paddingVertical: 20,
		paddingHorizontal:16,

		borderBottomColor: 'gray',
		borderBottomWidth: 1,
	},
	backButton: {
		// width: 100,
		// height: 30,
		width: 24,
		height: 24,
		justifyContent: 'center',
		alignItems: 'center',
	},
	title: {
		fontSize: 20,
	},
});
