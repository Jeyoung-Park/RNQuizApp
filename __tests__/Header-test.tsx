import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/shared/components/Header';

it('renders correctly', () => {
	const tree = renderer.create(
		<Header
			title="Quiz"
			goBack={() => console.log('go Back')}
			goBackText="Go Back"
		/>,
	);
	expect(tree).toMatchSnapshot();
});
