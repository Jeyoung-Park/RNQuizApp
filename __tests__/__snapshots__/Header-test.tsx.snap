// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 20,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
  >
    <Text>
      Go Back
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Quiz
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 30,
        "justifyContent": "center",
        "width": 100,
      }
    }
  />
</View>
`;
